lib/monitoring/monitoringClientInterface.js

```diff
@@ -0,0 +1,137 @@
+const Client = require('prom-client');
+
+/**
+ * Wrapper for Prometheus client - https://github.com/siimon/prom-client
+ */
+
+class ClientWrapper {
+
+    constructor() {
+        this.client = Client;
+    }
+
+    /**
+     * Gets all the metrics in the form of a string for prometheus server
+     * @return {Object} Registers the metrics
+     */
+    getMetrics() {
+        return this.client.register.metrics();
+    }
+
+    /**
+     * Counters go up, and reset when the process restarts.
+     * @param {string} metricsName - name of the metrics
+     * @param {string} metricsHelp - description of metrics
+     * @param {string} metricsLabels - Static labels applied to every metric
+     * @return {Object}
+     * Optional Parameter - metricsLabels
+     */
+    createCounter(metricsName, metricsHelp, metricsLabels) {
+        let label;
+        if (metricsLabels === undefined) {
+            label = [];
+        } else {
+            label = metricsLabels;
+        }
+        return new this.client.Counter({
+            name: metricsName,
+            help: metricsHelp,
+            labels: label,
+        });
+    }
+
+    /**
+     * Gauges are similar to Counters but Gauges value can be decreased.
+     * @param {string} metricsName - name of the metrics
+     * @param {string} metricsHelp - description of metrics
+     * @param {string} metricsLabels - Static labels applied to every metric
+     * @return {Object}
+     * Optional Parameter - metricsLabels
+     */
+    createGauge(metricsName, metricsHelp, metricsLabels) {
+        let label;
+        if (metricsLabels === undefined) {
+            label = [];
+        } else {
+            label = metricsLabels;
+        }
+        return new this.client.Gauge({
+            name: metricsName,
+            help: metricsHelp,
+            labels: label,
+        });
+    }
+
+    /**
+     * Histograms track sizes and frequency of events.
+     * The defaults buckets are intended to cover usual web/rpc requests,
+     * this can however be overriden.
+     * @param {string} metricsName - name of the metrics
+     * @param {string} metricsHelp - description of metrics
+     * @param {string} metricsBuckets - size of buckets for cumulative counters
+     * @param {string} metricsLabels - Static labels applied to every metric
+     * @return {Object}
+     * Optional Parameter - metricsBuckets and metricsLabels
+     */
+    createHistogram(metricsName, metricsHelp, metricsBuckets, metricsLabels) {
+        let label;
+        let bucket;
+        if (metricsBuckets === undefined) {
+            bucket =
+            [0.005, 0.01, 0.025, 0.05, 0.1, 0.25, 0.5, 1, 2.5, 5, 10];
```

Comment:
Why do defaults from `prom-client` need to be duplicated here?

Reply:
Because we are supplying a parameter, it cannot be empty.

Reply:
If the user does not supply the buckets parameter, then an empty variable will be sent.